javatests/arcs/android/host/AndroidSdkHandleManagerTest.kt

```diff
@@ -0,0 +1,262 @@
+package arcs.android.host
+
+import android.app.Application
+import androidx.lifecycle.Lifecycle
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.work.testing.WorkManagerTestInitHelper
+import arcs.android.storage.handle.AndroidHandleManager
+import arcs.core.data.FieldType
+import arcs.core.data.Schema
+import arcs.core.data.SchemaFields
+import arcs.core.data.SchemaName
+import arcs.core.host.HandleMode
+import arcs.core.host.SdkHandleManager
+import arcs.core.storage.driver.RamDiskStorageKey
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.sdk.ReadWriteCollection
+import arcs.sdk.ReadWriteSingleton
+import arcs.sdk.ReadableCollection
+import arcs.sdk.ReadableSingleton
+import arcs.sdk.WritableCollection
+import arcs.sdk.WritableSingleton
+import arcs.sdk.android.storage.service.DefaultConnectionFactory
+import arcs.sdk.android.storage.service.testutil.TestBindingDelegate
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.coroutines.experimental.suspendCoroutine
+
+typealias Person = TestParticleInternal1
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+@RunWith(AndroidJUnit4::class)
+class AndroidSdkHandleManagerTest {
```

Comment:
As you'd mentioned, these tests are pretty monolithic. Perhaps it's worth capturing while it's fresh in your mind a granular list (in a comment, or in a filed issue) of the unit tests you'd write right now if you had more time, so that someone can field soon following and we don't forget or miss things.

Reply:
We can port the SIngletonImplTest/CollectionImpltests that were deleted. They are micro tests of basic functionality. 

Not tested here is clear(), remove()